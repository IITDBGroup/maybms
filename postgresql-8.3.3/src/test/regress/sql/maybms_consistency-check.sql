
--Test for joins on uncertain quereis. The tuples with the same variables but
--different domain values should not be joined

create table r (a int, b int);

insert into r values (1,1),(1,2),(2,1),(2,2);

create table s as repair key a in r;

select s1.a as a1, s1.b as b1, s2.a as a2, s2.b as b2 from s s1, s s2;

select possible s1.a as a1, s1.b as b1, s2.a as a2, s1.b as b1, s2.b as b2 from s s1, s s2;

create table t as
select s1.a as a1, s1.b as b1, s2.a as a2, s2.b as b2 from s s1, s s2;

select a1, b1, a2, b2, s.a as a3, s.b as b3 from t, s;

select possible a1, b1, a2, b2, s.a as a3, s.b as b3 from t, s;

drop table r;
drop table s;
drop table t;
