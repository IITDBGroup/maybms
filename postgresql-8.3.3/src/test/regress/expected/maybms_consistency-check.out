--Test for joins on uncertain quereis. The tuples with the same variables but
--different domain values should not be joined
create table r (a int, b int);
insert into r values (1,1),(1,2),(2,1),(2,2);
create table s as repair key a in r;
select s1.a as a1, s1.b as b1, s2.a as a2, s2.b as b2 from s s1, s s2;
 a1 | b1 | a2 | b2 | _v0 | _d0 | _p0 | _v1 | _d1 | _p1 
----+----+----+----+-----+-----+-----+-----+-----+-----
  1 |  1 |  1 |  1 |   1 |   1 | 0.5 |   1 |   1 | 0.5
  1 |  1 |  2 |  1 |   1 |   1 | 0.5 |   2 |   4 | 0.5
  1 |  1 |  2 |  2 |   1 |   1 | 0.5 |   2 |   3 | 0.5
  1 |  2 |  1 |  2 |   1 |   2 | 0.5 |   1 |   2 | 0.5
  1 |  2 |  2 |  1 |   1 |   2 | 0.5 |   2 |   4 | 0.5
  1 |  2 |  2 |  2 |   1 |   2 | 0.5 |   2 |   3 | 0.5
  2 |  1 |  1 |  1 |   2 |   4 | 0.5 |   1 |   1 | 0.5
  2 |  1 |  1 |  2 |   2 |   4 | 0.5 |   1 |   2 | 0.5
  2 |  1 |  2 |  1 |   2 |   4 | 0.5 |   2 |   4 | 0.5
  2 |  2 |  1 |  1 |   2 |   3 | 0.5 |   1 |   1 | 0.5
  2 |  2 |  1 |  2 |   2 |   3 | 0.5 |   1 |   2 | 0.5
  2 |  2 |  2 |  2 |   2 |   3 | 0.5 |   2 |   3 | 0.5
(12 rows)

select possible s1.a as a1, s1.b as b1, s2.a as a2, s1.b as b1, s2.b as b2 from s s1, s s2;
 a1 | b1 | a2 | b1 | b2 
----+----+----+----+----
  1 |  1 |  1 |  1 |  1
  1 |  1 |  2 |  1 |  1
  1 |  1 |  2 |  1 |  2
  1 |  2 |  1 |  2 |  2
  1 |  2 |  2 |  2 |  1
  1 |  2 |  2 |  2 |  2
  2 |  1 |  1 |  1 |  1
  2 |  1 |  1 |  1 |  2
  2 |  1 |  2 |  1 |  1
  2 |  2 |  1 |  2 |  1
  2 |  2 |  1 |  2 |  2
  2 |  2 |  2 |  2 |  2
(12 rows)

create table t as
select s1.a as a1, s1.b as b1, s2.a as a2, s2.b as b2 from s s1, s s2;
select a1, b1, a2, b2, s.a as a3, s.b as b3 from t, s;
 a1 | b1 | a2 | b2 | a3 | b3 | _v0 | _d0 | _p0 | _v1 | _d1 | _p1 | _v2 | _d2 | _p2 
----+----+----+----+----+----+-----+-----+-----+-----+-----+-----+-----+-----+-----
  1 |  1 |  1 |  1 |  1 |  1 |   1 |   1 | 0.5 |   1 |   1 | 0.5 |   1 |   1 | 0.5
  1 |  1 |  2 |  1 |  1 |  1 |   1 |   1 | 0.5 |   2 |   4 | 0.5 |   1 |   1 | 0.5
  1 |  1 |  2 |  2 |  1 |  1 |   1 |   1 | 0.5 |   2 |   3 | 0.5 |   1 |   1 | 0.5
  2 |  1 |  1 |  1 |  1 |  1 |   2 |   4 | 0.5 |   1 |   1 | 0.5 |   1 |   1 | 0.5
  2 |  1 |  2 |  1 |  1 |  1 |   2 |   4 | 0.5 |   2 |   4 | 0.5 |   1 |   1 | 0.5
  2 |  2 |  1 |  1 |  1 |  1 |   2 |   3 | 0.5 |   1 |   1 | 0.5 |   1 |   1 | 0.5
  2 |  2 |  2 |  2 |  1 |  1 |   2 |   3 | 0.5 |   2 |   3 | 0.5 |   1 |   1 | 0.5
  1 |  2 |  1 |  2 |  1 |  2 |   1 |   2 | 0.5 |   1 |   2 | 0.5 |   1 |   2 | 0.5
  1 |  2 |  2 |  1 |  1 |  2 |   1 |   2 | 0.5 |   2 |   4 | 0.5 |   1 |   2 | 0.5
  1 |  2 |  2 |  2 |  1 |  2 |   1 |   2 | 0.5 |   2 |   3 | 0.5 |   1 |   2 | 0.5
  2 |  1 |  1 |  2 |  1 |  2 |   2 |   4 | 0.5 |   1 |   2 | 0.5 |   1 |   2 | 0.5
  2 |  1 |  2 |  1 |  1 |  2 |   2 |   4 | 0.5 |   2 |   4 | 0.5 |   1 |   2 | 0.5
  2 |  2 |  1 |  2 |  1 |  2 |   2 |   3 | 0.5 |   1 |   2 | 0.5 |   1 |   2 | 0.5
  2 |  2 |  2 |  2 |  1 |  2 |   2 |   3 | 0.5 |   2 |   3 | 0.5 |   1 |   2 | 0.5
  1 |  1 |  1 |  1 |  2 |  1 |   1 |   1 | 0.5 |   1 |   1 | 0.5 |   2 |   4 | 0.5
  1 |  1 |  2 |  1 |  2 |  1 |   1 |   1 | 0.5 |   2 |   4 | 0.5 |   2 |   4 | 0.5
  1 |  2 |  1 |  2 |  2 |  1 |   1 |   2 | 0.5 |   1 |   2 | 0.5 |   2 |   4 | 0.5
  1 |  2 |  2 |  1 |  2 |  1 |   1 |   2 | 0.5 |   2 |   4 | 0.5 |   2 |   4 | 0.5
  2 |  1 |  1 |  1 |  2 |  1 |   2 |   4 | 0.5 |   1 |   1 | 0.5 |   2 |   4 | 0.5
  2 |  1 |  1 |  2 |  2 |  1 |   2 |   4 | 0.5 |   1 |   2 | 0.5 |   2 |   4 | 0.5
  2 |  1 |  2 |  1 |  2 |  1 |   2 |   4 | 0.5 |   2 |   4 | 0.5 |   2 |   4 | 0.5
  1 |  1 |  1 |  1 |  2 |  2 |   1 |   1 | 0.5 |   1 |   1 | 0.5 |   2 |   3 | 0.5
  1 |  1 |  2 |  2 |  2 |  2 |   1 |   1 | 0.5 |   2 |   3 | 0.5 |   2 |   3 | 0.5
  1 |  2 |  1 |  2 |  2 |  2 |   1 |   2 | 0.5 |   1 |   2 | 0.5 |   2 |   3 | 0.5
  1 |  2 |  2 |  2 |  2 |  2 |   1 |   2 | 0.5 |   2 |   3 | 0.5 |   2 |   3 | 0.5
  2 |  2 |  1 |  1 |  2 |  2 |   2 |   3 | 0.5 |   1 |   1 | 0.5 |   2 |   3 | 0.5
  2 |  2 |  1 |  2 |  2 |  2 |   2 |   3 | 0.5 |   1 |   2 | 0.5 |   2 |   3 | 0.5
  2 |  2 |  2 |  2 |  2 |  2 |   2 |   3 | 0.5 |   2 |   3 | 0.5 |   2 |   3 | 0.5
(28 rows)

select possible a1, b1, a2, b2, s.a as a3, s.b as b3 from t, s;
 a1 | b1 | a2 | b2 | a3 | b3 
----+----+----+----+----+----
  1 |  1 |  1 |  1 |  1 |  1
  1 |  1 |  1 |  1 |  2 |  1
  1 |  1 |  1 |  1 |  2 |  2
  1 |  1 |  2 |  1 |  1 |  1
  1 |  1 |  2 |  1 |  2 |  1
  1 |  1 |  2 |  2 |  1 |  1
  1 |  1 |  2 |  2 |  2 |  2
  1 |  2 |  1 |  2 |  1 |  2
  1 |  2 |  1 |  2 |  2 |  1
  1 |  2 |  1 |  2 |  2 |  2
  1 |  2 |  2 |  1 |  1 |  2
  1 |  2 |  2 |  1 |  2 |  1
  1 |  2 |  2 |  2 |  1 |  2
  1 |  2 |  2 |  2 |  2 |  2
  2 |  1 |  1 |  1 |  1 |  1
  2 |  1 |  1 |  1 |  2 |  1
  2 |  1 |  1 |  2 |  1 |  2
  2 |  1 |  1 |  2 |  2 |  1
  2 |  1 |  2 |  1 |  1 |  1
  2 |  1 |  2 |  1 |  1 |  2
  2 |  1 |  2 |  1 |  2 |  1
  2 |  2 |  1 |  1 |  1 |  1
  2 |  2 |  1 |  1 |  2 |  2
  2 |  2 |  1 |  2 |  1 |  2
  2 |  2 |  1 |  2 |  2 |  2
  2 |  2 |  2 |  2 |  1 |  1
  2 |  2 |  2 |  2 |  1 |  2
  2 |  2 |  2 |  2 |  2 |  2
(28 rows)

drop table r;
drop table s;
drop table t;
